### ISSUE TITLE
TypeError: Property 'log' does not exist on type 'Config'

### ISSUE THREAD

**Issue by verdie-g** [tgriesser/knex#3007]
# Environment

Knex version: 0.16.3
Database + version: PostgreSQL 11.1
OS: up-to-date Arch Linux

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

As mentioned in the doc, the config object of knex can contain a "log" field but it is missing in the [type definition](https://github.com/tgriesser/knex/blob/master/types/knex.d.ts#L668).

2. Error message

```
TSError: ⨯ Unable to compile TypeScript:
src/repository/knex.ts(9,3): error TS2345: Argument of type '{ log: { debug: LeveledLogMethod; warn: LeveledLogMethod; error: LeveledLogMethod; }; debug: boolean; client?: string | typeof Client; dialect?: string; version?: string; connection?: string | ConnectionConfig | ... 4 more ... | SocketConnectionConfig; ... 8 more ...; asyncStackTraces?: boolean; }' is not assignable to parameter of type 'Config'.
  Object literal may only specify known properties, and 'log' does not exist in type 'Config'.
```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to mysql or postgresql, then single file example which initializes
   needed data and demonstrates the problem.

```ts
import * as knex from 'knex';

const db = knex({
  log: {
    warn: console.log,
    error: console.log,
    deprecate: console.log,
    debug: console.log,
  },
});
```

**Comment by JasperTack** [comment]
Is there any update about this? I have this problem as well

**Comment by jasonmacdonald** [comment]
Same problem. There doesn't appear to be a log config anymore but the docs says it should be there.

**Comment by robations** [comment]
I think this might have been resolved through #3148 and #3221 . Not sure why I didn't see your bug report first, I usually/always search for pre-existing bugs.

**Comment by kibertoad** [comment]
Should have been fixed by 0.17.0, please reopen if it is not.
